libs/cypress-indexeddb/src/lib/create-object-store.ts

```diff
@@ -0,0 +1,65 @@
+import { createVersionUpdateDatabaseConnection } from './upgrade-database';
+
+function createObjectStoreInternal(
+  database: IDBDatabase,
+  storeName: string,
+  isExistingStore: boolean
+): Promise<IDBObjectStore> {
+  let store: IDBObjectStore;
+
+  if (isExistingStore) {
+    store = database.transaction(storeName, 'readwrite').objectStore(storeName);
+  } else {
+    store = database.createObjectStore(storeName);
+  }
+
+  return new Promise<IDBObjectStore>((resolve, reject) => {
+    store.transaction.oncomplete = () => {
+      database.close();
+      resolve(store);
+    };
+    store.transaction.onerror = (e) => {
+      reject(e);
+    };
+  });
+}
+
+export function createObjectStore(
+  existingDatabase: IDBDatabase,
+  storeName: string
+): Promise<IDBObjectStore> {
+  let error: any;
+  const isExistingStore = existingDatabase.objectStoreNames.contains(storeName);
+  const log = Cypress.log({
+    name: `create`,
+    type: 'child',
+    message: `IDBObjectStore - ${storeName}`,
+    consoleProps: () => ({
+      'store name': storeName,
+      'did store exist before?': isExistingStore,
+      'database name': existingDatabase.name,
+      error: error || 'no',
+    }),
+    autoEnd: false,
+  });
+  if (existingDatabase?.constructor?.name !== 'IDBDatabase') {
```

Comment:
maybe we could reuse the function from above. Just extract it to a separate file

```suggestion
  if (!isIDBDatabase(existingDatabase)) {
```